Rename EditorTilesetFilter to MapEditorData

Ectras opened the pull request:
Closes #13857.
Renamed the EditorTilesetFilter class to MapEditorData (and EditorTilesetFilterInfo to MapEditorDataInfo).
Created an UpdateRule (RenameEditorTilesetFilter.cs). Updated all four mods with it.

Ectras commented:
Moved the UpdateRule to the Rules folder, filtered out the .ormap files

Mailaender commented on OpenRA.Mods.Common/OpenRA.Mods.Common.csproj:
```diff
@@ -919,22 +920,14 @@
-    <Copy
-      SourceFiles="$(TargetPath).mdb"
-      DestinationFolder="$(SolutionDir)mods/common/"
-      Condition="Exists('$(TargetPath).mdb')"/>
```
Please don't commit these changes.

Ectras commented:
I'm really sorry for that. I committed the original file now (with just the necessary changes). Hope that's ok.

pchote commented on OpenRA.Mods.Common/UpdateRules/Rules/RenameEditorTilesetFilter.cs:
```diff
@@ -0,0 +1,38 @@
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			foreach (var spawner in actorNode.ChildrenMatching("EditorTilesetFilterInfo"))
```
The `Info` classes are an internal implementation detail - they aren't exposed to the yaml, so you don't need to worry about them here :)

pchote commented on OpenRA.Mods.Common/UpdateRules/Rules/RenameEditorTilesetFilter.cs:
```diff
@@ -0,0 +1,38 @@
+			foreach (var spawner in actorNode.ChildrenMatching("EditorTilesetFilterInfo"))
+				spawner.RenameKey("MapEditorDataInfo");
+
+			foreach (var spawner in actorNode.ChildrenMatching("EditorTilesetFilter"))
```
Please rename this variable to `editor` or `data` or something similar (copy/paste leftover from `WormSpawner`).

pchote reviewed (commented):
Code changes look reasonable aside from two minor nits.  I'm not able to test this ingame.

pchote commented:
We'll also want you to squash the commits together once all the fixups are sorted

Ectras commented:
I feel a bit like a fool right now. Well, once again, really sorry. Fixed the two mistakes and squashed everything into one commit. Hope I didn't miss anything this time.

Ectras committed:
Renamed EditorTilesetFilter to MapEditorData and added an update rule

teinarss reviewed (approved):
I did some testing: 
* RenameEditorTilesetFilter works as intended
* No problems found in ActorSelectorLogic

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1254a393e5e50a54042866f406beccef39b32575...17205547d729143b585f552c4c572afdb66590c1)
